project/frontend/src/containers/CreateCompetition/CreateCompetition.js

```diff
@@ -24,7 +27,24 @@ const CreateCompetition = (props) => {
   const onParticipantDelete = (participant) =>
     setParticipantList(participantList.filter((v) => v !== participant?.email));
 
+  function checkValidDate() {
+    const start = new Date(startDate);
+    const end = new Date(endDate);
+    const now = new Date();
+    const today = new Date(now.getFullYear(), now.getMonth(), now.getDate());
+    if (end < start) {
+      setValidDate(-1);
+    } else if (start < today) {
+      setValidDate(-2);
+    } else {
+      setValidDate(1);
```

Comment:
These numbers are quite arbitrary - how about creating 2 state variables like `boolean dateIsValid` and `string dateErrorMessage`, then in the JSX returned you can just do
```js
{dateIsValid && (<p>{dateErrorMessage}</p>)}
```

Reply:
Thanks Tex very appreciate your advice. I removed the setValidDate state and just have a dateErrorMessage state instead. Nothing will be displayed when the date is valid. 0382f33900ff176da3424186bd77435f4be2366a